tests/test_esm.py

```diff
@@ -174,6 +186,8 @@ def test_disable_esm(self):
         # the keyring file is removed
         keyring_file = self.trusted_gpg_dir / 'ubuntu-esm-keyring.gpg'
         self.assertFalse(keyring_file.exists())
+        # credentials are removed
+        self.assertEqual(self.apt_auth_file.read_text(), '')
```

Comment:
I think you should test that only our entry is removed, and not that the file is empty. I.e., start the test with some dummy unrelated entry and make sure that at the end it's the only thing remaining.

Reply:
There's a separate test (`test_enable_fips_auth_if_other_entries`) for this

Reply:
That test doesn't check for removal, because there is no disabling of fips

Reply:
Ah, you're right, sorry. Fixed the test.